modules/gdscript/gdscript_compiler.cpp

```diff
@@ -355,6 +355,12 @@ GDScriptCodeGenerator::Address GDScriptCompiler::_parse_expression(CodeGen &code
 						r_error = ERR_COMPILATION_FAILED;
 						return GDScriptCodeGenerator::Address();
 					}
+
+					// Force script parsing if not done yet.
+					Script *script = Object::cast_to<Script>(res.ptr());
+					if (!script->is_valid()) {
+						script->reload();
+					}
```

Comment:
I don't understand why this would be needed. If the script is loaded by the resource loader it should already be fully compiled.

Reply:
This is what exactly happens in the reproduction project. Before reaching this part of the code (in _parse_expression > case GDScriptParser::Node::IDENTIFIER), it will go first to GDScriptCompiler::_gdtype_from_datatype > GDScriptCache::get_shallow_script, where it calls `script->set_path(p_path, true);` which push the uncompiled script inside the resource loader.
Another working solution is to change `get_shallow_script` to `get_full_script`, but it might have performance impact.

Reply:
Changed to get_full_script instead as you recommend. For the error handling, Changed to get_full_script instead as you recommend. For the error handling, maybe we could find a better text...